Felienne/hedy
# Implements commands language-switcher

**Description**
In this PR we implement the functionality to switch between languages within the code blocks. Each code blocks is implemented with an additional dropdown many iff the current user language supports translated keywords. At this moment English, Dutch and Spanish are supported. When using either Dutch or Spanish a dropdown menu is shown on each code block to support switching between English and the user language.

This implementation is created my making a dedicated `selector_container` on each code block. On selecting a language we send the `ace editor id`, `old language` and `target language` to `app.ts` for processing. A POST is sent to the back-end with the code from the editor to retrieve a keyword-translated version. Afterwards the two languages in the dropdown menu are switched (as the other language is now selected).

The starting `keyword_language` and `other_keyword_language` are stored in the window.State. As the syntax highlighter files for non-english languages also support the English keywords we initialize the syntax highlighter for each ace editor on `other_language`.

**Fix for**
This PR fixes #1577.

**How to test**
Notice the new blue button at the right-top of each code block. When selecting another language the keywords are translated accordingly. Verify that the copy button does copy the correct translated code. Also verify that you can switch back and forth between languages within the code editor.

**Checklist**

If you're unsure about any of these, don't hesitate to ask. We're here to help!
